corehq/apps/smsforms/app.py

```diff
@@ -135,6 +134,9 @@ def _clean_xml_for_partial_submission(xml, should_remove_case_actions):
     :param should_remove_case_actions: if True, remove case actions (create, update, close) from xml
     :return: byte str of cleaned xml
     """
+    if not xml:
+        raise PartialSubmissionXMLIsNone
```

Comment:
Do we need an a new type of exception for this since there is no known code path where this would happen, and it is not caught anywhere? An assertion would be good enough, although even that might not be necessary since there is no known way that this could happen.

When I wrote https://github.com/dimagi/commcare-hq/pull/31276#discussion_r832438554 there was a code path where `xml` could be `None` here, but I think since you've refactored it in fe6bbeb792caeca91716ed4572b450a2a9199f4f there is not. However, I might be wrong and there might be some other way that I missed for `None` to get in here--feel free to point that out.

Reply:
Yeah I knew I was removing the known code path, but figured it didn't hurt to have that test. Though it makes sense to draw the line at known code paths for tests/additional code, so I've removed this in fd6460155257de86f571919dd691afa81fd8637f. Thanks for flagging